CMakeLists.txt

```diff
@@ -638,6 +638,22 @@ if(CURL_BROTLI)
   endif()
 endif()
 
+option(CURL_ZSTD "Set to ON to enable building curl with zstd support." OFF)
+set(HAVE_ZSTD OFF)
+if(CURL_ZSTD)
+  find_package(Zstd REQUIRED)
```

Comment:
Probably too late to ask, but why relying on a custom CMake module file, while zstd installs an official CMake config file?

Reply:
If you think we can improve the cmake stuff for zstd, feel free to submit a pull-request!

Reply:
Now I remember: zstd provides config file since 1.4.5 (released in May of this year), so it makes sense to have a custom module file in order to properly handle previous versions.

@bagder Yep, when I have time, I'll see if there is a way to first check for zstd config file (more robust), then fallback to custom module file.

Reply:
Maybe use the same technique CMake does for curl: https://gitlab.kitware.com/cmake/cmake/-/blob/39677de5e209445c8cbc5957c1e79088d5d2a03a/Modules/FindCURL.cmake

So in the find module first check if zstd can be found in config mode. If not, do custom search and provide imported targets consistent with what zstd config currently produces.